packager/mpd/base/xml/xml_node.cc

```diff
@@ -387,10 +389,12 @@ bool RepresentationXmlNode::AddVODOnlyInfo(const MediaInfo& media_info) {
     RCHECK(AddChild(std::move(base_url)));
   }
 
-  const bool need_segment_base =
+
+  const bool need_segment_base = !use_segment_list && (
       media_info.has_index_range() || media_info.has_init_range() ||
-      (media_info.has_reference_time_scale() && !media_info.has_text_info());
-  DCHECK(!need_segment_base || !use_segment_list_text);
+      (media_info.has_reference_time_scale() && !media_info.has_text_info()));
```

Comment:
I suggest combining the need_segment_base and use_segment_list into one. It can reduce code duplications and simplify the code a lot.

```c++
const bool need_segment_base_or_list = use_segment_list ||
                                       use_segment_list_text ||
                                       media_info.has_index_range() ||
                                       media_info.has_init_range() ||
                                       media_info.has_reference_time_scale();
if (!need_segment_base_or_list)
  return true;

XmlNode child((use_segment_list || use_segment_list_text) ?
                   "SegmentList" : "SegmentBase");
...
```